Sources/XMLCoder/Box/ArrayBox.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  ArrayBox.swift
+//  XMLCoderPackageDescription
+//
+//  Created by Vincent Esche on 11/20/18.
+//
+
+import Foundation
+
+// Minimalist implementation of an order-preserving unkeyed box:
+internal class ArrayBox {
```

Comment:
Is there a specific reason for this to be a `class` instead of `struct`? Also I think this could conform to `Collection` or `Sequence` and get `map` and `filter` implementations for free from protocol extensions already available in the Swift standard library.